tests/lib/rules/jsx-key.js

```diff
@@ -37,7 +37,15 @@ ruleTester.run('jsx-key', rule, {
     {code: '[1, 2, 3].foo(x => <App />);'},
     {code: 'var App = () => <div />;'},
     {code: '[1, 2, 3].map(function(x) { return; });'},
-    {code: 'foo(() => <div />);'}
+    {code: 'foo(() => <div />);'},
+    // #1574
+    {code: 'React.Children.toArray([1, 2 ,3].map(x => <App />));'},
+    {
+      code: `
+      import { Children } from "react";
+      Children.toArray([1, 2 ,3].map(x => <App />));
```

Comment:
Let's also include a test case that has `const { toArray } = Children`, and another that has something called `toArray` and `Children` that aren't imported from React.

Reply:
Good idea, my changes wouldn't catch those cases. I'm also worried about other likely cases, this is a test case that I'm currently trying to get working as valid:

```js
import React from 'react';
import { Children as ReactChildren } from 'react';

const { Children } = React;
const { toArray } = Children;

React.Children.toArray([1, 2 ,3].map(x => <App />));
ReactChildren.toArray([1, 2 ,3].map(x => <App />));
Children.toArray([1, 2 ,3].map(x => <App />));
toArray([1, 2 ,3].map(x => <App />));
```

I need to get used to hunting down variable/import declarations. 😄

Reply:
One more question: should CommonJS also work?

```js
const { Children } = require('react');
```

Reply:
One more question, should CommonJS also work?

```js
const { Children } = require('react');
```

Reply:
Yes, definitely.

Reply:
ping @silvenon, are you interested in completing this PR?

Reply:
I am, but I don't have the skills for it yet, sorry 😥